library_registry/admin/controller.py

```diff
@@ -42,9 +50,54 @@ def log_in(self):
         else:
             return INVALID_CREDENTIALS
 
+    def log_in_with_token(self):
+        """JWT Token based login to create more RESTful API
+
+        Returns:
+            Response: With valid credentials this will return JWT Access and Refresh tokens as Set-Cookie headers and redirects to 'admin/admin_view
+            Response: With invalid credentials this will return the INVALID CREDENTIALS response
+        """
+        username = request.form.get("username")
+        password = request.form.get("password")
+        if Admin.authenticate(self._db, username, password):
+            access_token = create_access_token(identity=username)
+            refresh_token = create_refresh_token(identity=username)
+            response = make_response(
+                redirect(url_for('admin.admin_view')), 302)
+            set_access_cookies(response, access_token)
+            set_refresh_cookies(response, refresh_token)
+            return response
```

Comment:
We should consider how we need this returned (I think some conversations about this have already started). The log in/out/refresh methods should be able to interact with web, mobile and API clients, and to me the best way to do that would be to return the two tokens explicitly. But I'm sure there are factors I'm not considering!

Reply:
From what I understand of this, you are correct, this will require a bit of changes that would affect the front end too however.  We would need to remove redirects to explicitly send access and refresh tokens. And front ends relying on our current flow would be affected accordingly.

Reply:
yeah, that's the point I'm stuck on, I haven't been able to land on a satisfying solution that supports both front-ends. Let's discuss and see if we can figure something out.

Reply:
I think the satisfying solution is to combine both.  Please check how I have updated the `@after_request` decorator to specify for cookies.  I can do the same check for the explicit refresh end point as well to prevent cookie users from requesting explicitly.